lib/lightstep/tracer.rb

```diff
@@ -213,5 +220,27 @@ def extract_from_text_map(operation_name, carrier)
 
       span
     end
+
+    def inject_to_http_headers(span, carrier)
+      carrier[CARRIER_TRACER_STATE_PREFIX + 'spanid'] = span.span_context.id
+      carrier[CARRIER_TRACER_STATE_PREFIX + 'traceid'] = span.span_context.trace_id unless span.span_context.trace_id.nil?
+      carrier[CARRIER_TRACER_STATE_PREFIX + 'sampled'] = 'true'
+
+      span.span_context.baggage.each do |key, value|
+        carrier[CARRIER_BAGGAGE_PREFIX + key.gsub(/[^A-Za-z0-9-]/, '')] = value
```

Comment:
probably better to log and error and drop here, as the other side of the pipe isn't going to invert this transformation anyway.